cli/run_util.go

```diff
@@ -685,6 +692,9 @@ func activateAndRunSteps(
 	// ------------------------------------------
 	// Main - Preparing & running the steps
 	for idx, stepListItm := range workflow.Steps {
+		stepUniqueID := uuid.Must(uuid.NewV4()).String()
+		stepIDProperties := coreanalytics.Properties{"step_unique_id": stepUniqueID}
+		stepStartedProperties := workflowProperties.Merge(stepIDProperties)
```

Comment:
This is a bit hard to follow, but if I understand this correctly, then the `step_finished` event will also contain all these properties, including the workflow properties. I don't think that's necessary, we only need the step UUID, right? Given the volume of data we'll generate, I wouldn't include the unnecessary workflow properties in the `step_finished` event.